crates/uv-build-backend/src/fs_write_dispatcher.rs

```diff
@@ -0,0 +1,263 @@
+//! Dispatcher between writing to a directory, writing to a zip, writing to a `.tar.gz` and
+//! listing files.
+
+use crate::wheel::{write_hashed, RecordEntry};
+use crate::Error;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use fs_err::File;
+use sha2::{Digest, Sha256};
+use std::io::{BufReader, Cursor, Write};
+use std::path::{Path, PathBuf};
+use std::{io, mem};
+use tar::{EntryType, Header};
+use tracing::trace;
+use uv_fs::Simplified;
+use zip::{CompressionMethod, ZipWriter};
+
+/// Dispatcher between writing to a directory, writing to a zip, writing to a `.tar.gz` and
+/// listing files.
+///
+/// All paths are string types instead of path types since wheels are portable between platforms.
+///
+/// Contract: You must call close before dropping to obtain a valid output (dropping is fine in the
+/// error case).
+pub(crate) trait FsWriteDispatcher {
```

Comment:
This is an awfully uninformative name. It's like a virtual fs, but append-only and it's actually more our sink than a real fs.